chromadb/utils/embedding_functions.py

```diff
@@ -92,3 +105,91 @@ def __init__(self, model_name: str = "hkunlp/instructor-base", device= "cpu"):
 
     def __call__(self, texts: Documents) -> Embeddings:
         return self._model.encode(texts).tolist()
+
+
+#### CHROMA DEFAULT ONNX EMEBEDDING FUNCTION ####
+# In order to remove dependencies on sentence-transformers, which in turn depends on pytorch and sentence-piece we have
+# created a default ONNX embedding function that implements the same functionality as "all-MiniLM-L6-v2" from sentence-transformers.
+# visit https://github.com/chroma-core/onnx-embedding for the source code to generate and verify the ONNX model.
+
+# Borrow'ed from https://gist.github.com/yanqd0/c13ed29e29432e3cf3e7c38467f42f51
+# Download with tqdm to preserve the sentence-transformers experience
+def download(url: str, fname: Path, chunk_size=1024):
+    resp = requests.get(url, stream=True)
+    total = int(resp.headers.get('content-length', 0))
+    with open(fname, 'wb') as file, tqdm(
+        desc=str(fname),
+        total=total,
+        unit='iB',
+        unit_scale=True,
+        unit_divisor=1024,
+    ) as bar:
+        for data in resp.iter_content(chunk_size=chunk_size):
+            size = file.write(data)
+            bar.update(size)
+
+# Use pytorches default epsilon for division by zero
+# https://pytorch.org/docs/stable/generated/torch.nn.functional.normalize.html
+def normalize(v):
+    norm = np.linalg.norm(v)
+    if norm == 0:
+        norm = 1e-12
+    return v / norm
+
+
+class DefaultOnnxEmbeddingFunction(EmbeddingFunction):
```

Comment:
Agreed - will change

Reply:
Although to be honest I don't think we can safely every change the default since it implies breaking backwards compatibility

Reply:
You're right. This is yet another issue with not having a good way to store EFs currently. Let's note that it might be a breaking change, but go ahead with it.

Reply:
So in this case it's not a breaking change, the new model is 1:1 with the old but future ones may not be. I'll modify the code such that it's extensible but just noting that future changes will be backwards incompatible